runtime/src/accounts_db.rs

```diff
@@ -537,9 +541,14 @@ impl AccountsDB {
         inc_new_counter_info!("clean-old-root-reclaim-ms", measure.as_ms() as usize);
     }
 
-    fn clear_uncleaned_roots(&self) {
-        let mut accounts_index = self.accounts_index.write().unwrap();
-        accounts_index.uncleaned_roots.clear();
+    fn reset_uncleaned_roots(&self) {
+        let previous_roots = {
+            let mut accounts_index = self.accounts_index.write().unwrap();
```

Comment:
This converts a hashset into a vec which can be done outside the accounts_index write lock.

Reply:
Nice catch! Well, I didn't need to cast to `Vec<Slot>` first of all..